### PR TITLE
runtime-rs: Add default configuration file for cloud-hypervisor

### PR DESCRIPTION
**Pull request by amshinde**
The config template file for clh is in the new format for runtime-rs. It is a result of merging the new format file and options supportted by cloud-hypervisor.

Some config options from the golang runtime are missing as they may not be currently supported by the rust runtime. An example of this is the selinux options, rate limiting options as these are not currently supported or verified with the rust runtime.

Fixes: #8249

### CODE REVIEW

File: src/runtime-rs/config/configuration-clh.toml.in [hunk 1]
```diff
@@ -0,0 +1,352 @@
+# Copyright (c) 2019-2022 Alibaba Cloud
```
Comment (jodh-intel) [inline src/runtime-rs/config/configuration-clh.toml.in:1]: Possibly bike-shedding, but I'm in two minds about the `clh` part of the filename:

- `clh` matches the golang filename convention of referring to Cloud Hypervisor as `clh`, but the rust runtime internally mostly now uses `ch`.
- `clh` also matches the name of the hypervisor used by the golang runtime (`[hypervisor.clh]`), but with the rust runtime we're now using `[hypervisor.cloud-hypervisor]` for more clarity.

In summary, I'm wondering if we should call it `configuration-cloud-hypervisor.toml.in` since that would

- Match the contents of the file (the `hypervisor` name).
- Make it clear to users / packagers that this file is not of the same format as the golang runtime config.

**Review by jodh-intel — COMMENTED** [review]
Thanks @amshinde  - A couple of comments...

File: src/runtime-rs/config/configuration-clh.toml.in (hunk 1 above)
Comment (fidencio) [reply src/runtime-rs/config/configuration-clh.toml.in:1]: With the same spirit, if we're going to make it `cloud-hypervisor` we should follow the same pattern on the makefile vars as well.

File: src/runtime-rs/Makefile [hunk 2]
```diff
@@ -93,6 +93,9 @@ DEFAULT_HYPERVISOR ?= $(HYPERVISOR_DB)
 ##VAR HYPERVISOR=<hypervisor_name> List of hypervisors this build system can generate configuration for.
 HYPERVISORS := $(HYPERVISOR_DB) $(HYPERVISOR_ACRN) $(HYPERVISOR_FC) $(HYPERVISOR_QEMU) $(HYPERVISOR_CLH)
 
+CLHPATH := $(CLHBINDIR)/$(CLHCMD)
```
[the remark this answers is not in the document]
Comment (amshinde) [reply src/runtime-rs/Makefile:97]: Fixed this @jodh-intel

File: src/runtime-rs/config/configuration-clh.toml.in (hunk 1 above)
Comment (amshinde) [reply src/runtime-rs/config/configuration-clh.toml.in]: @jodh-intel I am still in two minds about this one. Also depends on if we eventually converge on a single format for the config. For now, I have renamed the `configuration-clh.toml.in` file to `configuration-cloud-hypervisor.toml.in`.  I have held off on renaming the variables in the makefile as these are not really user-facing.

Comment (jodh-intel) [reply src/runtime-rs/config/configuration-clh.toml.in]: I've just realised that we **cannot** to use `clh` for this version of the CH config file since when it is packaged, we need a unique name as all the config files end up in the same `./opt/kata/share/defaults/kata-containers/` in the archive.

Hence, `cloud-hypervisor` wfm :smile:

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+#   This limitation could be a bug in the kernel
+# Default number of bridges per SB/VM:
+# unspecified or 0   --> will be set to @DEFBRIDGES@
+# > 1 <= 5           --> will be set to the specified number
+# > 5                --> will be set to 5
+default_bridges = @DEFBRIDGES@
+
+# Block storage driver to be used for the hypervisor in case the container
+# rootfs is backed by a block device. This is virtio-blk.
+block_device_driver = "virtio-blk"
```
Comment (jodh-intel) [inline src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:165]: This needs to be `virtio-blk-pci` (or `virtio-blk-mmio`) as runtime-rs changed the values for this variable.

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
```
Comment (jodh-intel) [inline src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:166]: Can you add a `vm_rootfs_driver=`. See https://github.com/kata-containers/kata-containers/blob/main/src/runtime-rs/config/configuration-dragonball.toml.in#L27..L29

**Review by jodh-intel — COMMENTED** [review]
Thanks @amshinde - some more comment.

Also, as a point of note is that the golang `disable_selinux=` option doesn't exist in the rust runtime so is not specified in the rust CH config file.

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in (hunk 3 above)
Comment (amshinde) [reply src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:165]: Will change this then. I think we need to add some validation in the runtime-rs for this.

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in (hunk 4 above)
Comment (amshinde) [reply src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:166]: Its already added.

Comment (amshinde) [reply src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:166]: I had added this for parity, but I was wondering why not just use the `block_device_driver` option here. Why have an additional `vm_rootfs_driver` ?

**Comment by amshinde** [comment]
> Thanks @amshinde - some more comment.
> 
> Also, as a point of note is that the golang `disable_selinux=` option doesn't exist in the rust runtime so is not specified in the rust CH config file.

Yes. Summarazing the list of missing options here:
`rootless=`, `disable_selinux=`, `disable_seccomp=`, `guest_selinux_label=`, `disable_guest_empty_dir=` and all the disk+network rate limiter options. 
Additionally for the runtime, missing options include `vfio_mode=`, `disable_guest_empty_dir=`, `experimental=` , `enable_pprof=`, all tracings options and options related to image downloading for confidential containers.

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+# rootfs filesystem type:
+#   - ext4 (default)
+#   - xfs
+#   - erofs
+rootfs_type=@DEFROOTFSTYPE@
+ 
+# Block storage driver to be used for the VM rootfs is backed
+# by a block device.
+vm_rootfs_driver = "@VMROOTFSDRIVER_CLH@"
```
Comment (jodh-intel) [inline src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:25]: This is being set to `"virtio-blk-pci"` but it needs to be set to `"virtio-pmem"` by the build.

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+# For example, use `kernel_params = "vsyscall=emulate"` if you are having
+# trouble running pre-2.15 glibc.
+#
+# WARNING: - any parameter specified here will take priority over the default
+# parameter value of the same name used to start the virtual machine.
+# Do not set values here unless you understand the impact of doing so as you
+# may stop the virtual machine from booting.
+# To see the list of default parameters, enable hypervisor debug, create a
+# container and look for 'default-kernel-parameters' log entries.
+kernel_params = "@KERNELPARAMS@"
```
Comment (jodh-intel) [inline src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:83]: This is being set to:

```toml
kernel_params = "console=ttyS1 agent.log_vport=1025"
```

... but that makes the container boot fail. I think you'll need to make `$KERNELPARAMS` hypervisor-specific / dragonball-only as for CH, this needs to expand to and empty string, as we do for the golang runtime:

```toml
kernel_params = ""
```

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in (hunk 4 above)
Comment (jodh-intel) [reply src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:164]: See: #7139.

**Review by jodh-intel — COMMENTED** [review]
Thanks @amshinde - A few more comments.

Please could you resolve and test the generated config?

**Comment by jodh-intel** [comment]
@amshinde and @GabyCT - see #8306.

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in (hunk 5 above)
Comment (amshinde) [reply src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:25]: I had tested this with virtio-blk-pci and it worked for me. I suppose the virtio-pmem requirement is for TDX?

File: src/runtime-rs/config/configuration-cloud-hypervisor.toml.in (hunk 6 above)
Comment (amshinde) [reply src/runtime-rs/config/configuration-cloud-hypervisor.toml.in:83]: Ok. I'll look into this, but I tested the the generated config few days back and it worked at my end.

**Review by jodh-intel — APPROVED** [review]
Thanks @amshinde - Tested in default, DM and TDX scenarios with minor local tweaks for paths and suitable config tweaks for testing with TDX.

lgtm

**Review by dborquez — APPROVED** [review]
